Client/Frontend/Home/BookmarksPanel.swift

```diff
@@ -188,14 +159,15 @@ class BookmarksPanel: SiteTableViewController, HomePanel {
     func tableView(tableView: UITableView, editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [AnyObject]? {
         if source == nil {
             return [AnyObject]()
-        } else if source!.current.count == 0 && source!.current.guid == BookmarkRoots.MobileFolderGUID {
+        }
+
+        if !(source?.current.itemIsEditableAtIndex(indexPath.row) ?? false) {
```

Comment:
Move this check up to https://github.com/mozilla/firefox-ios/pull/852/files#diff-3a015464c840b339778d834499cc82d4R152. This delegate is what tells the table view if we want edit actions or not so we don't need to return an empty array for nothing.